src/__mocks__/isomorphic-fetch.ts

```diff
@@ -1,4 +1,18 @@
-const Fetch = (url: string, options: any): Promise<any> => {
+import fetch from "isomorphic-fetch"; //tslint:disable-line
+
+const Fetch = (url: string, options: RequestInit): Promise<any> => {
+  if (options.body!.toString().includes("non-error-class")) {
+    return Promise.reject({
```

Comment:
I can't recall, but if I recall 400 or bad status doesn't trigger a promise rejection. I think server timeout or 500 does

Reply:
```
The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.
By default, fetch won't send or receive any cookies from the server, resulting in unauthenticated requests if the site relies on maintaining a user session (to send cookies, the credentials init option must be set).
Since Aug 25, 2017. The spec changed the default credentials policy to same-origin. Firefox changed since 61.0b13.
```